### PR TITLE
[WIP] Fix orc reader RLEv2 reader

### LINKED ISSUE #5324: [BUG] read_orc reads incorrect data on one row
**Describe the bug**
When reading data via `cudf.read_orc` one of the rows seems to have incorrect data.

**Steps/Code to reproduce bug**
```python
import cudf

df = cudf.read_orc("row_bug.orc")
df.iloc[3874]

_col0     10100163
_col1         4125
_col2    637000238
Name: 3874, dtype: int32
```

**Expected behavior**
```python
import pyarrow.orc as orc
pdf = orc.ORCFile("row_bug.orc").read().to_pandas()
pdf.iloc[3874]

_col0     10100163
_col1         4125
_col2    927000494
Name: 3874, dtype: int32
```
Value of `_col2` at index location `3874` should be `927000494`

**Environment overview (please complete the following information)**
 - Method of cuDF install: Conda
   - 0.14 nightly on May 29

**Additional context**
Data here: [row_bug.orc.zip](https://github.com/rapidsai/cudf/files/4703780/row_bug.orc.zip)

Thanks @trstovall for pointing out the issue.

cc: @OlivierNV

### LINKED ISSUE #5440: [BUG] cudf.read_orc reads incorrect data for one row
**Describe the bug**

**Steps/Code to reproduce bug**
```
>>> df = cudf.read_orc('to_orc_bug.orc')
>>> df.upc_nbr[(df.visit_nbr == 14600028) & (df.store_nbr == 47)] = 681131184420
>>> df.to_orc('to_orc_bug.orc', compression='snappy')
>>> df2 = cudf.read_orc('to_orc_bug.orc')
>>> df2.upc_nbr[(df2.visit_nbr == 14600028) & (df2.store_nbr == 47)]
999786    2526351652
```

**Expected behavior**
Returned value should be `681131184420`, not `2526351652`.

**Environment overview (please complete the following information)**
 - Method of cuDF install: Conda
 - 0.14 nightly on ~ May 29

[to_orc_bug.orc.zip](https://github.com/rapidsai/cudf/files/4761005/to_orc_bug.orc.zip)

### PR DESCRIPTION
**Pull request by devavret**
Fixes #5440 
Fixes #5324 

Opened [ORC-642](https://issues.apache.org/jira/browse/ORC-642) to track issue in apache's docs.

### CODE REVIEW

File: cpp/src/io/orc/stripe_data.cu [hunk 1]
```diff
@@ -642,6 +642,18 @@ static const __device__ __constant__ uint8_t kRLEv2_W[32] = {
...
   17, 18, 19, 20, 21, 22, 23, 24, 26, 28, 30, 32, 40, 48, 56, 64};
 
+/**
+ * @brief Maps the RLEv2 patch size (pw + pgw) to number of bits
+ *
+ * Patch size (in bits) is only allowed to be from the below set. If `pw + pgw == 34` then the size
+ * of the patch in the file is the smallest size in the set that can fit 34 bits i.e.
+ * `ClosestFixedBitsMap[34] == 40`
+ */
+static const __device__ __constant__ uint8_t ClosestFixedBitsMap[65] = {
```
Comment (vuule) [inline cpp/src/io/orc/stripe_data.cu:652]: Is it possible to initialize this with a function so that the rules are evident in the code?

Comment (devavret) [reply cpp/src/io/orc/stripe_data.cu:652]: This is taken from https://github.com/apache/orc/blob/master/c%2B%2B/src/RLEV2Util.cc#L30-L37 so I don't actually know the rule. The set of fixed bit widths and how they're calculated is not in the docs.

Comment (vuule) [reply cpp/src/io/orc/stripe_data.cu:652]: Then it's fine as an array. Would be great if you could include a link to the source in some way.

**Comment by harrism** [comment]
Great job and thanks for the fix, @devavret

**Pull request merged**